---
title: "Untitled"
output: html_document
---

```{bash}

source activate qiime2-2019.7

name=call19
mapper=call_mapper.tsv

qiime feature-table filter-samples \
  --i-table table-$name"".qza \
  --m-metadata-file $mapper \
  --p-where "[sample_type]='feces'" \
  --o-filtered-table table-$name""-fecal.qza
  
qiime feature-table summarize \
  --i-table table-$name""-fecal.qza \
  --o-visualization table-$name""-fecal.qzv \
  --m-sample-metadata-file $mapper
  
qiime diversity alpha-rarefaction \
  --i-table table-$name""-fecal.qza \
  --i-phylogeny rooted-tree-$name"".qza \
  --p-max-depth 4000 \
  --m-metadata-file $mapper \
  --o-visualization alpha-rarefaction.qzv

  
bash qiime_figTwo.sh $name""-fecal $name 1000 $mapper
bash qiime_figTwo.sh $name""-fecal $name 4000 $mapper
bash qiime_figTwo.sh $name""-fecal $name 800 $mapper
bash qiime_figTwo.sh $name""-fecal $name 3000 $mapper

# filter out wolbachia reads
qiime taxa filter-table \
  --i-table table-$name""-fecal.qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Wolbachia \
  --o-filtered-table table-$name""-fecal_noW.qza
  
qiime feature-table summarize \
  --i-table table-$name""-fecal_noW.qza \
  --o-visualization table-$name""-fecal_noW.qzv \
  --m-sample-metadata-file $mapper
  
bash qiime_figTwo.sh $name""-fecal_noW $name 1000 $mapper
bash qiime_figTwo.sh $name""-fecal_noW $name 4000 $mapper
bash qiime_figTwo.sh $name""-fecal_noW $name 700 $mapper
bash qiime_figTwo.sh $name""-fecal_noW $name 3000 $mapper

qiime feature-table filter-samples \
  --i-table table-$name""-fecal_noW.qza \
  --m-metadata-file $mapper \
  --p-where "[genotype_class]='mutant'" \
  --o-filtered-table table-$name""-fecal_noW-mutants.qza
  
qiime feature-table core-features \
  --i-table table-$name""-fecal_noW-mutants.qza \
  --o-visualization core_features-$name""-fecal_noW-mutants.qzv
  
qiime feature-table filter-samples \
  --i-table table-$name""-fecal_noW.qza \
  --m-metadata-file $mapper \
  --p-where "[genotype_class]='control'" \
  --o-filtered-table table-$name""-fecal_noW-control.qza
  
qiime feature-table core-features \
  --i-table table-$name""-fecal_noW-control.qza \
  --o-visualization core_features-$name""-fecal_noW-control.qzv
  
qiime diversity alpha \
  --i-table table-$name""-fecal_noW.qza \
  --p-metric "shannon" \
  --o-alpha-diversity table-$name""-fecal_noW-alpha_shannon 
  
qiime tools export --input-path taxonomy-$name"".qza --output-path exported


#Goes back to figure 1 analysis
cd no_Wolbachia
name=gerald
mapper=gerald_mapper.tsv
qiime feature-table filter-samples \
  --i-table table-$name""_noW.qza \
  --m-metadata-file $mapper \
  --p-where "NOT [originalID]='g9.5-m'" \
  --o-filtered-table table-$name""_noW_noG.qza
  
# filter out archaea reads
qiime taxa filter-table \
  --i-table table-$name""_noW_noG.qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Archaea \
  --o-filtered-table table-$name""_noW_noG_noArch.qza
  
qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG_noArch.qza \
  --m-metadata-file $mapper \
  --p-where "[sex]='f'" \
  --o-filtered-table table-$name""_noW_noG_noArch_females.qza
  
qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG_noArch.qza \
  --m-metadata-file $mapper \
  --p-where "[sex]='m'" \
  --o-filtered-table table-$name""_noW_noG_noArch_males.qza
  
bash qiime_figOne.sh $name""_noW_noG_noArch_females $name 3000 $mapper
bash qiime_figOne.sh $name""_noW_noG_noArch_males $name 3000 $mapper

bash allAncoms.sh $name""_noW_noG_noArch_females $name $mapper
bash allAncoms.sh $name""_noW_noG_noArch_males $name $mapper

#splitting into files that can compare the genotypes 1 by 1

qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG.qza \
  --m-metadata-file $mapper \
  --p-where "NOT [Genotype]='w1118'" \
  --o-filtered-table table-$name""_noW_noG_no1118.qza
  
qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG.qza \
  --m-metadata-file $mapper \
  --p-where "NOT [Genotype]='park25 het'" \
  --o-filtered-table table-$name""_noW_noG_noHet.qza
  
qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG.qza \
  --m-metadata-file $mapper \
  --p-where "NOT [Genotype]='park25 homozygous'" \
  --o-filtered-table table-$name""_noW_noG_noHomo.qza

bash ancomComparison $name""_noW_noG_no1118 $name $mapper
bash ancomComparison $name""_noW_noG_noHet $name $mapper
bash ancomComparison $name""_noW_noG_noHomo $name $mapper

qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG.qza \
  --m-metadata-file $mapper \
  --p-where "[sex]='m'" \
  --o-filtered-table table-$name""_noW_noG_males.qza
  
qiime feature-table filter-samples \
  --i-table table-$name""_noW_noG.qza \
  --m-metadata-file $mapper \
  --p-where "[sex]='f'" \
  --o-filtered-table table-$name""_noW_noG_females.qza
  
  
bash ../qiime_figOne.sh $name""_noW_noG_males $name 3000 $mapper
bash ../qiime_figOne.sh $name""_noW_noG_females $name 3000 $mapper


table_name=$name""_noW_noG_females  #changed from male to female to do both tests because my file didn't like the test

qiime diversity adonis \
  --i-distance-matrix noG/females/core-metrics-results-$table_name""/unweighted_unifrac_distance_matrix.qza \
  --m-metadata-file $mapper \
  --p-formula "Genotype" \
  --o-visualization noG/females/unweighted_permanova-male-type-$table_name""

qiime diversity adonis \
  --i-distance-matrix noG/females/core-metrics-results-$table_name""/weighted_unifrac_distance_matrix.qza \
  --m-metadata-file $mapper \
  --p-formula "Genotype" \
  --o-visualization noG/females/weighted_permanova-male-type-$table_name""

qiime diversity adonis \
  --i-distance-matrix noG/females/core-metrics-results-$table_name""/bray_curtis_distance_matrix.qza \
  --m-metadata-file $mapper \
  --p-formula "Genotype" \
  --o-visualization noG/females/bray_permanova-male-type-$table_name""



cd ..
qiime feature-table filter-samples \
  --i-table table-$name"".qza \
  --m-metadata-file $mapper \
  --p-where "NOT [originalID]='g9.5-m'" \
  --o-filtered-table table-$name""_noG.qza

qiime taxa filter-table \
  --i-table table-$name""_noG.qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Archaea \
  --o-filtered-table table-$name""_noG_noArch.qza
  
bash qiime_figOne.sh $name""_noG $name 3000 $mapper
bash qiime_figOne.sh $name""_noG_noArch $name 3000 $mapper

```

```{r UCLA}
setwd('~/Dropbox/sequencing/Nov2019/')
file_path <- "UCLA"
mapper_file <- "call_mapper.tsv